projects/distributed-tracing/src/shared/dashboard/widgets/waterfall/waterfall/marker-tooltip/marker-tooltip.component.ts

```diff
@@ -0,0 +1,56 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'ht-marker-tooltip',
+  styleUrls: ['./marker-tooltip.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="marker-tooltip-container" *ngIf="this.data | async as data">
+      <div class="tooltip-header">
+        <div class="log-count">
+          Logs
+          <div class="count">{{ data.relativeTimes.length }}</div>
+        </div>
+        <div class="time-range" *ngIf="data.relativeTimes.length > 0">
+          <ng-container
+            *ngIf="data.relativeTimes[0] === data.relativeTimes[data.relativeTimes.length - 1]; else rangeResult"
+            >{{ data.relativeTimes[0] }}ms</ng-container
+          >
+          <ng-template #rangeResult
+            >{{ data.relativeTimes[0] }}ms - {{ data.relativeTimes[data.relativeTimes.length - 1] }}ms</ng-template
+          >
+        </div>
+      </div>
+      <div class="divider"></div>
+      <div
```

Comment:
are we ok with the visual presentation on this one? we should consider using a grid here. I think it'd be better if the keys and values are aligned evenly.

Reply:
Done